// amazeui-react/lib/Alert.js
'use strict';

var React = require('react');
var classNames = require('classnames');
var ClassNameMixin = require('./mixins/ClassNameMixin');

var Alert = React.createClass({displayName: "Alert",
  mixins: [ClassNameMixin],

  propTypes: {
    classPrefix: React.PropTypes.string.isRequired,
    amStyle: React.PropTypes.oneOf(['secondary', 'success', 'warning',
      'danger']),
    onClose: React.PropTypes.func
  },

  getDefaultProps: function() {
    return {
      classPrefix: 'alert'
    };
  },

  renderCloseBtn: function() {
    return (
      React.createElement("button", {
        type: "button", 
        className: this.setClassNamespace('close'), 
        onClick: this.props.onClose}, 
        "×"
      )
    );
  },

  render: function() {
    var classSet = this.getClassSet();
    var isCloseable = !!this.props.onClose;

    if (this.props.amStyle) {
      classSet[this.prefixClass(this.props.amStyle)] = true;
    }

    classSet[this.prefixClass('closeable')] = isCloseable;

    return (
      React.createElement("div", React.__spread({}, 
        this.props, 
        {className: classNames(this.props.className, classSet)}), 
        isCloseable ? this.renderCloseBtn() : null, 
        this.props.children
      )
    );
  }
});

module.exports = Alert;
