// amazeui-react/lib/Header.js
'use strict';

var React = require('react');
var classNames = require('classnames');
var ClassNameMixin = require('./mixins/ClassNameMixin');
var Icon = require('./Icon');

var Header = React.createClass({displayName: "Header",
  mixins: [ClassNameMixin],

  propTypes: {
    classPrefix: React.PropTypes.string,
    theme: React.PropTypes.oneOf(['default']),
    data: React.PropTypes.object,
    fixed: React.PropTypes.bool,
    title: React.PropTypes.node,
    link: React.PropTypes.string
  },

  getDefaultProps: function() {
    return {
      classPrefix: 'header',
      theme: 'default'
    };
  },

  renderTitle: function() {
    return this.props.title ? (
      React.createElement("h1", {className: this.prefixClass('title')}, 
        this.props.link ? (
          React.createElement("a", {
            href: this.props.link}, 
            this.props.title
          )
        ) : this.props.title
      )
    ) : null;
  },

  renderNav: function(position) {
    var data = this.props.data;
    var renderItem = function(item, i) {
      return (
        React.createElement("a", {href: item.link, 
           key: 'headerNavItem' + i}, 
          item.title ? (
            React.createElement("span", {className: this.prefixClass('nav-title')}, 
              item.title
            )
          ) : null, 

          item.customIcon ? (
            React.createElement("img", {src: item.customIcon, alt: item.title || null})
          ) : item.icon ? (
            React.createElement(Icon, {
              className: this.prefixClass('icon'), 
              icon: item.icon})
          ) : null
        )
      );
    }.bind(this);

    return data && data[position] ? (
      React.createElement("div", {
        className: classNames(this.prefixClass('nav'),
        this.prefixClass(position))}, 
        data[position].map(function(item, i) {
          return renderItem(item, i);
        })
      )
    ) : null;
  },

  render: function() {
    var classSet = this.getClassSet();

    // am-header-fixed: fixed header
    classSet[this.prefixClass('fixed')] = this.props.fixed;

    return (
      React.createElement("header", React.__spread({}, 
        this.props, 
        {"data-am-widget": this.props.classPrefix, 
        className: classNames(this.props.className, classSet)}), 
        this.renderNav('left'), 
        this.renderTitle(), 
        this.renderNav('right')
      )
    );
  }
});

module.exports = Header;
