// amazeui-react/lib/Navbar.js
'use strict';

var React = require('react');
var classNames = require('classnames');
var ClassNameMixin = require('./mixins/ClassNameMixin');
var Icon = require('./Icon');
var omit = require('object.omit');

var Navbar = React.createClass({displayName: "Navbar",
  mixins: [ClassNameMixin],

  propTypes: {
    classPrefix: React.PropTypes.string,
    theme: React.PropTypes.oneOf(['default']),
    data: React.PropTypes.array,
    onSelect: React.PropTypes.func
  },

  getDefaultProps: function() {
    return {
      classPrefix: 'navbar',
      theme: 'default',
      data: [],
      onSelect: function() {}
    };
  },

  render: function() {
    var classSet = this.getClassSet();
    var props = omit(this.props, 'data');

    return (
      React.createElement("div", React.__spread({}, 
        props, 
        {"data-am-widget": this.props.classPrefix, 
        cf: true, 
        className: classNames(this.props.className, classSet)}), 
        React.createElement("ul", {className: this.prefixClass('nav')}, 
          this.props.data.map(function(item, i) {
            return (
              React.createElement("li", {key: i, 
                  onClick: this.props.onSelect.bind(this, item.link)}, 
                React.createElement("a", {href: item.link}, 
                  item.customIcon ? (
                    React.createElement("img", {src: item.customIcon, alt: item.title})
                  ) : item.icon ? (
                    React.createElement(Icon, {icon: item.icon})
                  ) : null, 

                  item.title ? (
                    React.createElement("span", {className: this.prefixClass('label')}, 
                    item.title
                  )
                  ) : null
                )
              )
            );
          }.bind(this))
        )
      )
    );
  }
});

module.exports = Navbar;
