// amazeui-react/lib/NavItem.js
'use strict';

var React = require('react');
var classNames = require('classnames');
var ClassNameMixin = require('./mixins/ClassNameMixin');

var NavItem = React.createClass({displayName: "NavItem",
  mixins: [ClassNameMixin],

  propTypes: {
    active: React.PropTypes.bool,
    disabled: React.PropTypes.bool,
    header: React.PropTypes.bool,
    divider: React.PropTypes.bool,
    href: React.PropTypes.any,
    componentTag: React.PropTypes.node.isRequired
  },

  getDefaultProps: function() {
    return {
      classPrefix: 'nav',
      componentTag: 'li'
    };
  },

  render: function() {
    var classes = this.getClassSet();
    var props = this.props;
    var Component = props.componentTag;

    // del am-nav
    classes[this.setClassNamespace(props.classPrefix)] = false;

    // set classes
    classes[this.prefixClass('header')] = props.header;
    classes[this.prefixClass('divider')] = props.divider;

    if (props.href) {
      return this.renderAnchor(classes);
    }

    return (
      React.createElement(Component, React.__spread({}, 
        props, 
        {className: classNames(classes, props.className)}), 
          this.props.children
      )
    );
  },

  renderAnchor: function(classes) {
    var Component = this.props.componentTag;

    var linkProps = {
      href: this.props.href,
      title: this.props.tilte,
      target: this.props.target
    };

    return (
      React.createElement(Component, React.__spread({}, 
        this.props, 
        {className: classNames(classes, this.props.className)}), 
        React.createElement("a", React.__spread({},  linkProps), 
          this.props.children
        )
      )
    );
  }
});

module.exports = NavItem;

// TODO: DropDown Tab 处理
//       disabled 没有禁止链接跳转
