// amazeui-react/lib/Titlebar.js
'use strict';

var React = require('react');
var classNames = require('classnames');
var ClassNameMixin = require('./mixins/ClassNameMixin');
var AvgGrid = require('./AvgGrid');
var omit = require('object.omit');

var Titlebar = React.createClass({displayName: "Titlebar",
  mixins: [ClassNameMixin],

  propTypes: {
    classPrefix: React.PropTypes.string,
    theme: React.PropTypes.oneOf(['default', 'multi', 'cols']),
    nav: React.PropTypes.array,
    title: React.PropTypes.node
  },

  getDefaultProps: function() {
    return {
      classPrefix: 'titlebar',
      theme: 'default',
      data: []
    };
  },

  render: function() {
    var classSet = this.getClassSet();
    var props = omit(this.props, ['classPrefix', 'nav', 'theme']);

    return (
      React.createElement("div", React.__spread({}, 
        props, 
        {"data-am-widget": this.props.classPrefix, 
        className: classNames(this.props.className, classSet)}), 
        React.createElement("h2", {className: this.prefixClass('title')}, 
          this.props.href ? (
            React.createElement("a", {href: this.props.href}, 
              this.props.title
            )
          ) : this.props.title
        ), 
        this.props.nav ? (
          React.createElement("nav", {className: this.prefixClass('nav')}, 
            this.props.nav.map(function(item, i) {
              return (
                React.createElement("a", {href: item.link, key: i}, 
                  item.title
                ));
            })
          )
        ) : null
      )
    );
  }
});

module.exports = Titlebar;
