// amazeui-react/lib/UCheck.js
'use strict';

/**
 * Custom radio/checkbox style
 */

var React = require('react');
var classNames = require('classnames');
var ClassNameMixin = require('./mixins/ClassNameMixin');
var Input = require('./Input');
var constants = require('./constants');

var UCheck = React.createClass({displayName: "UCheck",
  mixins: [ClassNameMixin],

  propTypes: {
    type: React.PropTypes.oneOf(['radio', 'checkbox']),
    disabled: React.PropTypes.bool,
    amStyle: React.PropTypes.oneOf(['secondary', 'success', 'warning',
      'danger']),
    inline: React.PropTypes.bool,
    hasFeedback: React.PropTypes.bool
  },

  getDefaultProps: function() {
    return {
      type: 'checkbox'
    };
  },

  render: function() {
    var classSet = {};

    classSet[this.setClassNamespace(this.props.type)] = !this.props.inline;
    classSet[this.setClassNamespace(this.props.type + '-inline')] =
      this.props.inline;

    if (this.props.amStyle) {
      classSet[this.setClassNamespace(this.props.amStyle)] = true;
    }

    return (
      React.createElement("label", {className: classNames(this.props.className, classSet)}, 
        React.createElement(Input, React.__spread({}, 
          this.props, 
          {ref: "field", 
          className: this.setClassNamespace('ucheck-checkbox'), standalone: true})), 

        React.createElement("span", {className: this.setClassNamespace('ucheck-icons')}, 
          React.createElement("i", {className: "am-icon-unchecked"}), 
          React.createElement("i", {className: "am-icon-checked"})
        ), 

        this.props.label
      )
    );
  }
});

module.exports = UCheck;

// TODO: replace icon with Icon component
